### DISCUSSION TITLE
Export all

### DISCUSSION THREAD

**Discussion by gkrwns7566** [jowilf/starlette-admin#261] [Q&A]
My ModelView settings are in default, and I found that the export feature always fetches only the 10 rows in the current page no matter what rows I select. Is there any way to export all the rows in the whole table?? Showing all rows returns a timeout since the table has a lot of rows, which cannot solve my problem. Do i have to make it in my own custom batch action?

**Comment by mrharpo** [comment]
I'm looking into the same thing, and it's possible if you can load all the records onto the page, using `page_size_options = [10, 25, 100, -1]`

This works until about 10k rows, where the page becomes extremely large and slows down substantially. 

Around 100k rows, the server is unable to complete the request before timing out.

It looks like the easiest way would be through a separate API function.
